src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -4201,31 +4212,58 @@ private void ThrowIfNull(LLVMValueRef entry)
                 builder.BuildCondBr(builder.BuildICmp(LLVMIntPredicate.LLVMIntEQ, NullRefFunction.GetParam(1), LLVMValueRef.CreateConstPointerNull(LLVMTypeRef.CreatePointer(LLVMTypeRef.Int8, 0)), "nullCheck"),
                     throwBlock, retBlock);
                 builder.PositionAtEnd(throwBlock);
-                MetadataType nullRefType = _compilation.NodeFactory.TypeSystemContext.SystemModule.GetType("System", "NullReferenceException");
+                
+                CreateAndThrowException(builder, "NullReferenceException", NullRefFunction);
```

Comment:
The other codegens use `ThrowHelpers` for this purpose. E.g. there is `ThrowHelpers.ThrowNullReferenceException`.

Would it be make sense to do the same for WASM?

Reply:
Thanks, I notice there is no `ThrowArithmeticException` there, but there is `ThrowOverflowException`.  Looking at some of the code that was used for this , https://github.com/dotnet/coreclr/blob/9b0a9fd623/tests/src/JIT/IL_Conformance/Old/Base/ckfinite.il, it is catching an `OverflowException` but if I look at https://docs.microsoft.com/en-us/dotnet/api/system.reflection.emit.opcodes.ckfinite?view=netcore-3.1 it says it should throw an `ArithmeticException`, which is the direction I want, but I'd need to add a helper method for that.   Should I do that or use an `OverflowException` ?

Reply:
> it says it should throw an ArithmeticException

`OverflowException` inherits from `ArithmeticException` so the docs are not wrong. It is ok to throw more specific exception that what the docs say. It may be nice to make the docs more accurate.

The behavior should match what CoreCLR does.

In any case, you can add these helpers as needed.